fix: add micromark parser to markdownlint config

MatheusBaldi opened the pull request:
In `markdownlint` `0.36.0`, the support for `micromark` returned. Adding it in the config object fixes the error described in [this issue](https://github.com/silvermine/markdownlint-rule-indent-alignment/issues/7)

The new `micromark` version changes some previously assumed parser behaviors. It causes some errors related to nested intentation in HTML snippets. Since currently there is no need to lint HTML, the tokens related to it are now ignored.

Introduces a new test verification to prevent an error where `htmlText` tokens cause other text tokens to appear to be indented.

MrMarCode commented on src/indent-alignment.js:
```diff
@@ -22,6 +26,14 @@ function iterate(tokens, nodeTypes, iterateeFn) {
+      // linting lines that include it. Note: Micromark parser currently has a weird
+      // behavior where multiline HTML tags are incorrectly classified as paragraph
+      // tokens.
+      if (token.type.includes('htmlText')) {
```
```suggestion
      if (token.type === 'htmlText') {
```

MrMarCode replied:
Let's start with this for consistency with our other check.

MatheusBaldi replied:
Oh, thanks, I forgot about this one

MrMarCode commented on src/indent-alignment.js:
```diff
@@ -22,6 +26,14 @@ function iterate(tokens, nodeTypes, iterateeFn) {
          skipTo = undefined;
       }
 
+      // `htmlText` tokens cause other text tokens to appear to be indented, so we avoid
+      // linting lines that include it. Note: Micromark parser currently has a weird
+      // behavior where multiline HTML tags are incorrectly classified as paragraph
+      // tokens.
```
```suggestion
      // If an HTML tag spans multiple lines, and is not considered a "starting condition"
      // it will be parsed as part of a paragraph instead of an HTML block. For example:
      // ```md
      // <a
      // href="">this text is inside a paragraph.</a>
      //
      // <a href="">this text is inside a block of HTML.</a>
      // ```
      // See "starting condition" for more details:
      // https://spec.commonmark.org/0.31.2/#html-blocks
      //
      // Since we want to avoid linting any HTML content, we skip any tokens that appear
      // on the same line as `htmlText`. Note that if we did not do this, any text before
      // or after `htmlText` could be inaccurately flagged by this rule.
```

Sorry, I've gone back and forth on this comment. Since we talked, I did some more digging and found a more accurate explanation for what is going on here. Micromark is acting within the bounds of the commonmark spec so this is not a bug. However, it does create a hardship on our rule. So this comment should answer questions a new developer may have, such as:
Why are we skipping htmlText? Why would htmlText appear if we're already skipping htmlFlow? What is causing this edge case?

MatheusBaldi replied:
This explan way better what is happening. Thanks!

MrMarCode commented on src/indent-alignment.js:
```diff
@@ -8,7 +8,11 @@ function traverse(tokens, nodeTypes, iterateeFn) {
+      // Avoid linting html tags (`htmlFlow`) because it would create too many edge cases
+      // due to the way micromark parses HTML tokens. Currently any tokens inside a
+      // `paragraph` token are considered siblings, so it's too hard to define which
+      // elements should be aligned or indented.
```
```suggestion
      // Avoid linting HTML blocks because the commonmark spec does not define the proper
      // parent/child structure that would be needed to validate blocks of html text. For
      // example, a div that's inside another div is still considered a sibling by the
      // commonmark standard. See also: https://spec.commonmark.org/0.31.2/#html-blocks
```

MrMarCode commented on src/indent-alignment.js:
```diff
@@ -33,9 +45,14 @@ function iterate(tokens, nodeTypes, iterateeFn) {
 }
 
-function findFirstTokenOfType(tokens, nodeTypes) {
+function findFirstTokenOfTypeToVisit(tokens, nodeTypes) {
```
I'm glad you thought to change the name because it's good to [be descriptive with function names](https://github.com/silvermine/silvermine-info/blob/master/coding-standards.md#naming-1). 

Does `toVisit` explain what's going on here? If a developer never read the implementation, would they know what this function did based on the name? How about something like `findFirstNonHTMLTokenOfType`?

MatheusBaldi replied:
I got this idea from a variable in the `iterate` function, so, indeed, I believe he would need to read the implementation until finding it to undestand what it refers to. Thanks for the suggestion.

MrMarCode commented on package-lock.json:
It seems like there are more changes in the package lock than are necessary. Could you try this:
* Revert all changes for `package.json` and `package-lock.json` and delete your node_modules folder.
* Run `nvm use` (to ensure your on the right version of node and npm) and then `npm i`
* Change the markdownlint version to 0.36.1 in package.json.
* Run `npm i` again.

See if that reduces the number of changes in package-lock. Thanks! :slightly_smiling_face:

MatheusBaldi replied:
It did reduce! I was on node 18 and an older version of npm, so that explains the amount of changes. Thank you.

MrMarCode reviewed (changes requested):
Thanks for working through this upgrade. Your hard work is appreciated!

Note: I usually don't test my own suggestions thoroughly. So if you could always be sure to test changes before finishing up, even if they are based on suggestions, that would be appreciated as well!

MatheusBaldi committed:
fix: add micromark parser to markdownlint config

In markdownlint 0.36.0, the support for micromark returned. Adding it in the config
object fixes the error described in
[this issue](https://github.com/silvermine/markdownlint-rule-indent-alignment/issues/7)

The new micromark version changes some previously assumed parser behaviors. It causes
some errors related to nested intentation in HTML snippets. Since currently there is no
need to lint HTML, the tokens related to it are now ignored.

Introduces a new test verification to prevent an error where  tokens cause
other text tokens to appear to be indented.

MrMarCode reviewed (approved):
LGTM, thanks Matheus! :slightly_smiling_face:
